codalab/apps/web/models.py

```diff
@@ -533,32 +531,25 @@ def get_top_three(self):
                     # Getting next phase
                     next_phase = phases[index + 1]
                 break
-
-        if current_phase is not None:
+        if current_phase and current_phase is not None:
             local_scores = current_phase.scores()
-
-            for group in local_scores:
-                for _, scoredata in group['scores']:
-                    sub = CompetitionSubmission.objects.get(pk=scoredata['id'])
-                    scoredata['date'] = sub.submitted_at
-                    scoredata['count'] = sub.phase.submissions.filter(participant=sub.participant).count()
-
-            top_three = []
-
-            if len(local_scores) > 0 and len(local_scores[0]) > 0 and len(local_scores[0]['scores']) > 0:
-                num_part = len(local_scores[0]['scores'])
-                local_scores = local_scores[0]['scores']
-
-                if num_part > 3: # Keep us in 1-3 range.
-                    num_part = 3
-
-                for index in range(num_part):
-                    temp_dict = {
-                        'username': local_scores[index][1]['username'],
-                        'score': local_scores[index][1]['values'][0]['val']
-                    }
-                    top_three.append(temp_dict.copy())
-            return top_three # Return only our top 3, with the data we want.
+            main_score_def = None
+            formatted_score_list = list()
+            for score_dict in local_scores:
+                header_list = score_dict['headers']
+                main_score_def = header_list[0]
```

Comment:
will add a comment explaining what/why is happening here? maybe a large comment above this block?